Assets/Plugins/iOS/Shopify/BuyTests/BuySerializableTests.swift

```diff
@@ -0,0 +1,207 @@
+    //
+    func createPersonName() -> PersonNameComponents {
+        var personName        = PersonNameComponents.init()
+        personName.givenName  = firstName
+        personName.familyName = lastName
+        return personName
+    }
+    
+    func createPostalAddress() -> CNPostalAddress {
+        let postalAddress        = CNMutablePostalAddress.init()
+        postalAddress.street     = address1
+        postalAddress.postalCode = zip
+        postalAddress.country    = country
+        postalAddress.state      = province
+        postalAddress.city       = city
+        return postalAddress
+    }
+    
+    func createMultiPostalAddress() -> CNPostalAddress {
+        let postalAddress    = createPostalAddress() as! CNMutablePostalAddress
+        postalAddress.street = address1 + "\n" + address2
+        return postalAddress
+    }
+    
+    func createMultiPostalAddressEdgeCase() -> CNPostalAddress {
+        let postalAddress    = createPostalAddress() as! CNMutablePostalAddress
+        postalAddress.street = edgeAddress1 + "\n" + edgeAddress2
+        return postalAddress
+    }
+    
+    func createContact() -> PKContact {
+        let contact           = PKContact.init()
+        contact.name          = createPersonName()
+        contact.emailAddress  = emailAddress
+        contact.phoneNumber   = CNPhoneNumber.init(stringValue: phone)
+        contact.postalAddress = createPostalAddress()
+        return contact
+    }
+    
+    // ----------------------------------
+    //  MARK: - Convenience Asserts -
+    //
+    func assertEqualContact(_ contactDict: Dictionary<String, Any>, contact: PKContact) {
+        XCTAssertEqual(contactDict[PKContactSerializedField.firstName.rawValue] as? String, contact.name?.givenName);
+        XCTAssertEqual(contactDict[PKContactSerializedField.lastName.rawValue]  as? String, contact.name?.familyName);
+        XCTAssertEqual(contactDict[PKContactSerializedField.city.rawValue]      as? String, contact.postalAddress?.city);
+        XCTAssertEqual(contactDict[PKContactSerializedField.country.rawValue]   as? String, contact.postalAddress?.country);
+        XCTAssertEqual(contactDict[PKContactSerializedField.province.rawValue]  as? String, contact.postalAddress?.state);
+        XCTAssertEqual(contactDict[PKContactSerializedField.zip.rawValue]       as? String, contact.postalAddress?.postalCode);
+        XCTAssertEqual(contactDict[PKContactSerializedField.email.rawValue]     as? String, contact.emailAddress);
+        
+        if let firstAddress = contactDict[PKContactSerializedField.address1.rawValue] as? String {
+            if let secondAddress = contactDict[PKContactSerializedField.address2.rawValue] as? String {
+                XCTAssertEqual(firstAddress + "\n" + secondAddress, contact.postalAddress?.street);
+            }
+            else
+            {
+                XCTAssertEqual(firstAddress, contact.postalAddress?.street);
+            }
+        }
```

Comment:
Please make this assertion of address components to be explicit.

So if you're expecting 2 address components then assert those two components.

If you're expecting 1 address component ensure that there aren't two address components.